### PR TITLE
Migrate Debian based builds to bookworm

### PR DESCRIPTION
**Pull request by tobim**
Since bookworm is now in hard freeze this upgrade should be safe. This will allow us to bump the minimum version of GCC to 12 should the need arise.

### CODE REVIEW

**Comment by tobim** [comment]
CI should run now after the Apache JFrog issues are resolved.
The only remaining question is whether we should bump the minimum requirement for GCC already, since 10 will not be tested anymore once this PR is merged.

File: .github/workflows/vast.yaml
Comment (dominiklohmann) [inline .github/workflows/vast.yaml:781]: I think we should update the table in the docs in build.md already to gcc12+ and clang17+, as those are now the only versions we actively test with.

**Review by dominiklohmann — APPROVED** [review]
<3

File: Dockerfile
Comment (tobim) [reply Dockerfile:102]: It is still at 8 in nixpkgs because nobody bothered to bump it for spdlog yet.

File: .github/workflows/vast.yaml
Comment (patszt) [reply .github/workflows/vast.yaml:781]: Nice, i am actually struggling with my code not working on gcc 10 due to a bug. I am trying to update to gcc 11 myself. Can't wait for it to get merged :D

File: cmake/VASTRegisterPlugin.cmake
Comment (dominiklohmann) [inline cmake/VASTRegisterPlugin.cmake:165]: The same workaround needs to be appended to `VAST_FIND_DEPENDENCY_LIST` right below.

**Pull request merged**
